pkg/loop/plugin.go

```diff
@@ -0,0 +1,168 @@
+}
+
+// GRPCClient implements [plugin.GRPCPlugin] and returns a [ChainPlugin].
+func (p *grpcChainPlugin) GRPCClient(ctx context.Context, broker *plugin.GRPCBroker, conn *grpc.ClientConn) (interface{}, error) {
+	return &chainPluginClient{broker: broker, grpc: pb.NewPluginClient(conn), lggr: p.lggr}, nil
+}
+
+var _ ChainPlugin = (*chainPluginClient)(nil)
+
+type chainPluginClient struct {
+	grpc   pb.PluginClient
+	broker *plugin.GRPCBroker
+	lggr   logger.Logger
+}
+
+func (p *chainPluginClient) NewRelayer(ctx context.Context, config string, keystore Keystore) (types.Relayer, error) {
+	id := p.broker.NextId()
+	lis, err := p.broker.Accept(id)
+	if err != nil {
+		return nil, fmt.Errorf("broker failed to accept connection: %w", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterKeystoreServer(s, &keystoreServer{impl: keystore})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			p.lggr.Errorf("error serving Keystore connection %d: %w", id, err)
+		}
+	}()
+	reply, err := p.grpc.NewRelayer(ctx, &pb.NewRelayerRequest{
+		Config:     config,
+		KeystoreID: id,
+	})
+	if err != nil {
+		return nil, err
+	}
+	cc, err := p.broker.Dial(reply.RelayerID)
+	if err != nil {
+		return nil, fmt.Errorf("broker failed to dial connection %d: %w", reply.RelayerID, err)
+	}
+	return &relayerClient{p.broker, pb.NewRelayerClient(cc)}, nil
+}
+
+type chainPluginServer struct {
+	pb.UnimplementedPluginServer
+
+	impl   ChainPlugin
+	broker *plugin.GRPCBroker
+	lggr   logger.Logger
+}
+
+func (p *chainPluginServer) NewRelayer(ctx context.Context, request *pb.NewRelayerRequest) (*pb.NewRelayerReply, error) {
+	cc, err := p.broker.Dial(request.KeystoreID)
+	if err != nil {
+		return nil, fmt.Errorf("broker failed to dial connection: %w", err)
+	}
+	r, err := p.impl.NewRelayer(ctx, request.Config, &keystoreClient{pb.NewKeystoreClient(cc)})
+	if err != nil {
+		return nil, err
+	}
+	id := p.broker.NextId()
+	lis, err := p.broker.Accept(id)
+	if err != nil {
+		return nil, fmt.Errorf("broker failed to accept connection %d: %w", id, err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterRelayerServer(s, newRelayerServer(r, p.broker, p.lggr))
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			p.lggr.Errorf("error serving Relayer connection %d: %w", id, err)
+		}
+	}()
+	return &pb.NewRelayerReply{RelayerID: id}, nil
+}
+
+var _ Keystore = (*keystoreClient)(nil)
+
+type keystoreClient struct {
+	grpc pb.KeystoreClient
+}
+
+func (k *keystoreClient) Sign(ctx context.Context, account string, data []byte) ([]byte, error) {
+	reply, err := k.grpc.Sign(ctx, &pb.SignRequest{Account: account, Data: data})
+	if err != nil {
+		return nil, err
+	}
+	return reply.Signed, nil
+}
+
+var _ pb.KeystoreServer = (*keystoreServer)(nil)
+
+type keystoreServer struct {
+	pb.UnimplementedKeystoreServer
+
+	impl Keystore
+}
+
+func (k *keystoreServer) Sign(ctx context.Context, request *pb.SignRequest) (*pb.SignReply, error) {
+	signed, err := k.impl.Sign(ctx, request.Account, request.Data)
```

Comment:
Is the keystoreServer intended to be implemented in the client (plugin host), e.g. the core node?  Or, is this something that would be called by OCR (for example) and is chain specific?

Reply:
The `keystoreServer` runs on the host side, and packages up the existing two-step keystore interface of `Get(id)`ing a `Key` and then calling `Sign([]byte)` that the tx manager currently uses. Solana happens to be quite generic, but our keystore APIs are technically still chain-specific, so it's still an open question whether this pure byte signing API could work for every chain. The generic tx manager work is addressing this more directly.
```go
type SolanaKeystore interface {
    Get(id string) (Key, error)
}
type Key interface {
    Sign(msg []byte) ([]byte, error)
}
```

Reply:
That's because the keystore does more than store/get keys, it handles signing as well, correct?  Making it chain specific (or, at the very least, chains often do this differently)